Jenkinsfile.integration

```diff
@@ -222,7 +222,7 @@ pipeline {
         }
         cleanup {
             script {
-                    sh '[ -f /tmp/${SOCOK8S_ENVNAME}.needcleanup ] && ./run.sh teardown && rm /tmp/${SOCOK8S_ENVNAME}.needcleanup'
```

Comment:
Isn't it correct to remove the file **only** if the cleanup (teardown) succeeded? I don't understand what you try to fix here.

Reply:
If you have a docs only run, no network is created, so the teardown will fail and the file wont be cleaned up

Reply:
Im guessing we could fix this based on the vars somehow

Reply:
Good point. Don't modify the order. The "[ -f /tmp/${SOCOK8S_ENVNAME}.needcleanup ]" should be sufficient to skip teardown (i.e. remove the flag creation for docs). And if not, maybe introduce a new flag.

Reply:
> If you have a docs only run, no network is created, so the teardown will fail and the file wont be cleaned up

But if no network is created, the file is also not created.

Reply:
Actually I think the check is wrong as it'll exit non-zero which Jenkins considers a failure.

The correct fix, I think, is here: https://github.com/SUSE-Cloud/socok8s/pull/663

However, if the teardown still fails then the file won't be removed so this reorder may still be useful. But in those cases actually having a needcleanup file hang around could be useful for us to see what didn't get fully cleaned.

Reply:
In some conversations with @evrardjp a simple idea emerged, why don't keep the this file inside the workspace? Therefore if the CI eventually removes this workspace will remove this file too.

Reply:
Because removing the Jenkins workspace doesn't mean that the engcloud resources have been properly torn down.

Reply:
Yes, of course not. I agree with that. However I think the problem were are solving is how the needcleanup will be eventually removed (and where is stored) to prevent garbage in the CI. 
I understand if the task is tagged as functional then the CI will complete all the process, and finally, the teardown will be executed cleaning the resources. The condition prevents to execute the teardown (because fails) only in case of non-functional execution

Reply:
Right, so either the teardown will be successful, after which the `.needcleanup` will be removed, or something fails during the teardown at which point having a `.needcleanup` hang around can be a useful indication to us.

In terms of garbage on the CI, it's a file in /tmp, so I'm not too concerned about it. Just my 2cents :-).

Reply:
@jhesketh it will also be hard to track. Here everything is in the namespace.

For the test, I agree it could be made more userfriendly, as things will report a failure on docs run, which is improvable.

@ilausuch Can you make it a
```
if test -f ./.needcleanup; then
  ./run.sh teardown
fi
```

or alternatively, for the Jenkins wizards:

```
if (fileExists("/tmp/${env.SOCOK8S_ENVNAME}.needcleanup")) {
  sh './run.sh teardown'
}
```

Reply:
> @jhesketh it will also be hard to track. Here everything is in the namespace.
> 
> For the test, I agree it could be made more userfriendly, as things will report a failure on docs run, which is improvable.
> 

I disagree as twe can put the SOCOK8S_ENVNAME in the file.

Reply:
Oh yeah I missed that part. That's also fine for me. I just remember convos where we decided to put everything in workspace, and I thought it was backwards to put things outside namespace.